src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -1249,6 +1265,7 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		metadata.kernelspec = this._savedKernelInfo;
 		metadata.language_info = this.languageInfo;
 		metadata.tags = this._tags;
+		metadata.multi_connection_mode = this._multiConnectionMode ? this._multiConnectionMode : undefined;
```

Comment:
Are you purposely making it so that if the value is false it'll be set to undefined? Seems like you could just have it be `metadata.multi_connection_mode = this._multiConnectionMode;`

Reply:
Yeah - if it's undefined, the field won't be saved in the metadata at all since theres no use in saving multi_connection_mode: false.